### PR TITLE
Improve flagging guide & add flagging with Blocks

### PR DESCRIPTION
**Pull request by abidlabs**
I decided to improve the current Flagging guide rather than create a new one specifically for Blocks since I thought it would be more useful if the information was gathered in one place. 

Closes: #1675

### CODE REVIEW

File: guides/using_flagging.md [hunk 1]
```diff
@@ -112,7 +109,7 @@ iface = gr.Interface(
 
 iface.launch()
 ```
-<iframe src="https://hf.space/embed/aliabd/calculator-flagging-options/+" frameBorder="0" height="500" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
+<gradio-app space="gradio/calculator-flagging-options/">
```
Comment (freddyaboulton) [inline guides/using_flagging.md:112]: I think you need to add a closing tag here ` </gradio-app>`

File: guides/using_flagging.md [hunk 2]
```diff
@@ -139,23 +142,50 @@ iface = gr.Interface(
...
-<iframe src="https://hf.space/embed/aliabd/calculator-flagging-crowdsourced/+" frameBorder="0" height="500" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
 
-You can now see all the examples flagged above in this [public HF dataset](https://huggingface.co/datasets/aliabd/crowdsourced-calculator-demo/blob/main/data.csv).
+Notice that we define our own 
+instance of  `gradio.HuggingFaceDatasetSaver` using our Hugging Face token and
+the name of a dataset we'd like to save samples to. In addition, we also set `allow_flagging="manual"`
+because on Hugging Face Spaces, `allow_flagging` is set to `"never"` by default. Here's our demo:
+
+<gradio-app space="aliabd/calculator-flagging-crowdsourced/">
```
Comment (freddyaboulton) [inline guides/using_flagging.md:159]: I think this demo got moved to the gradio org

**Review by freddyaboulton — APPROVED** [review]
Awesome guide @abidlabs !!

Just think we need to close the `gradio-app` tag to get the demos to load.

**Comment by abidlabs** [comment]
Ah yes thanks @freddyaboulton!

**Comment by abidlabs** [comment]
Fixed both the closing tags and the naming of the Space

**Pull request merged**